enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/RevisionHistoryEnricher.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.builder.TypedVisitor;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentBuilder;
+import io.fabric8.kubernetes.api.model.extensions.DeploymentFluent;
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+/**
+ * This enricher adds the 'revisionHistoryLimit' property to deployment spec of RCs / RSs for KuberNetes/OpenShift resource descriptors.
+ * This property determines number of previous ReplicaControlller to retain in history in order to rollback previous one.
+ */
+
+public class RevisionHistoryEnricher extends BaseEnricher {
+
+    public static final String DEFAULT_NAME = "fmp-revision-history";
+    private static final String DEFAULT_NUMBER_OF_REVISIONS = "2";
+
+    // config keys
+    enum Config implements Configs.Key {
+        limit {{ d = DEFAULT_NUMBER_OF_REVISIONS; }};
+
+        protected String d;
+        public String def() { return d; }
+
+        public Integer toInt() {
+            return Integer.parseInt(d);
+        }
+    }
+
+    public RevisionHistoryEnricher(EnricherContext buildContext) {
+        super(buildContext, DEFAULT_NAME);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+        final Integer maxRevisionHistories = Integer.parseInt(getConfig(Config.limit));
```

Comment:
I suggest to add a `BaseEnricher.getConfigAsInt(...)` and remove the `toInt()` method on the Config enum.

Reply:
fixed

Reply:
ahaan but that's looks equivalent
Could you please let me thought behind `BaseEnricher.getConfigAsInt(...)`?  

Could you please me know why to remove `toInt()` method?

Please dont mind for this, I just want to understand thoughts behind it :)